nodebuilder/tests/swamp/swamp.go

```diff
@@ -120,15 +120,10 @@ func (s *Swamp) WaitTillHeight(ctx context.Context, height int64) libhead.Hash {
 		case <-ctx.Done():
 			require.NoError(s.t, ctx.Err())
 		case <-t.C:
-			status, err := s.ClientContext.Client.Status(ctx)
+			latest, err := s.ClientContext.LatestHeight()
 			require.NoError(s.t, err)
-
-			latest := status.SyncInfo.LatestBlockHeight
-			switch {
-			case latest == height:
-				return libhead.Hash(status.SyncInfo.LatestBlockHash)
-			case latest > height:
-				res, err := s.ClientContext.Client.Block(ctx, &height)
+			if latest >= height {
+				res, err := s.ClientContext.Client.Block(ctx, &latest)
```

Comment:
What's the goal here?

Reply:
simplifying the code - it's an unnecessary check.

Reply:
If this were production code, I would agree :D